src/material-experimental/mdc-slider/slider.spec.ts

```diff
@@ -0,0 +1,1449 @@
+
+    it('should set the value smaller than the min value', () => {
+      expect(sliderInstance.value).toBe(3);
+      expect(sliderInstance.min).toBe(4);
+      expect(sliderInstance.max).toBe(6);
+    });
+
+    it('should set the fill to the min value', () => {
+      expect(thumbContainerEl.style.transform).toContain('translateX(0px)');
+    });
+  });
+
+  describe('slider with set min and max and a value greater than max', () => {
+    let fixture: ComponentFixture<SliderWithValueSmallerThanMin>;
+    let sliderDebugElement: DebugElement;
+    let sliderNativeElement: HTMLElement;
+    let sliderInstance: MatSlider;
+    let thumbContainerEl: HTMLElement;
+
+    beforeEach(fakeAsync(() => {
+      fixture = createComponent(SliderWithValueGreaterThanMax);
+      fixture.detectChanges();
+
+      sliderDebugElement = fixture.debugElement.query(By.directive(MatSlider));
+      sliderNativeElement = sliderDebugElement.nativeElement;
+      sliderInstance = sliderDebugElement.componentInstance;
+      thumbContainerEl = <HTMLElement>sliderNativeElement
+        .querySelector('.mdc-slider__thumb-container');
+
+      // Flush the "requestAnimationFrame" timer that performs the initial
+      // rendering of the MDC slider.
+      flushRequestAnimationFrame();
+    }));
+
+    it('should set the value greater than the max value', () => {
+      expect(sliderInstance.value).toBe(7);
+      expect(sliderInstance.min).toBe(4);
+      expect(sliderInstance.max).toBe(6);
+    });
+
+    it('should set the fill to the max value', () => {
+      expect(thumbContainerEl.style.transform).toContain('translateX(100px)');
+    });
+  });
+
+  describe('slider with change handler', () => {
+    let fixture: ComponentFixture<SliderWithChangeHandler>;
+    let sliderDebugElement: DebugElement;
+    let sliderNativeElement: HTMLElement;
+    let testComponent: SliderWithChangeHandler;
+
+    beforeEach(() => {
+      fixture = createComponent(SliderWithChangeHandler);
+      fixture.detectChanges();
+
+      testComponent = fixture.debugElement.componentInstance;
+      spyOn(testComponent, 'onChange');
+      spyOn(testComponent, 'onInput');
+
+      sliderDebugElement = fixture.debugElement.query(By.directive(MatSlider));
+      sliderNativeElement = sliderDebugElement.nativeElement;
+    });
+
+    it('should emit change on mousedown', () => {
+      expect(testComponent.onChange).not.toHaveBeenCalled();
+
+      dispatchMousedownEventSequence(sliderNativeElement, 0.2);
+      fixture.detectChanges();
+
+      expect(testComponent.onChange).toHaveBeenCalledTimes(1);
+    });
+
+    it('should emit change on slide', () => {
+      expect(testComponent.onChange).not.toHaveBeenCalled();
+
+      dispatchSlideEventSequence(sliderNativeElement, 0, 0.4);
+      fixture.detectChanges();
+
+      expect(testComponent.onChange).toHaveBeenCalledTimes(1);
+    });
+
+    // TODO(devversion): MDC slider always emits change event on mouseup (regardless of value)
```

Comment:
Seems like a bug in MDC

Reply:
Issue submitted. Will add that to the comment: https://github.com/material-components/material-components-web/issues/5018